### PR TITLE
CLI: Deprecate `cpunode/gpunode/tpunode`, hide `admin`

### PR DESCRIPTION
**Pull request by concretevitamin**
Changes
- Deprecate + hide `cpunode/gpunode/tpunode`
  - Users/we have found them confusing, since `sky launch` essentially does the same thing without the final `ssh`
- Purge mentions of the above in docs
- Hide `admin`
- Minor rewording of docstrs

Now
```console
» sky gpunode                 
WARNING: `sky gpunode` is deprecated and will be removed in a future release. Please use `sky launch --gpus <gpus>` instead.

I 11-17 09:23:31 optimizer.py:694] == Optimizer ==
I 11-17 09:23:31 optimizer.py:706] Target: minimizing cost
I 11-17 09:23:31 optimizer.py:717] Estimated cost: $0.2 / hour
...

» sky                           
...
Commands:
  launch       Launch a cluster or task.
  exec         Execute a task or command on an existing cluster.
  status       Show clusters.
  cost-report  Show estimated costs for launched clusters.
  queue        Show the job queue for cluster(s).
  logs         Tail the log of a job.
  cancel       Cancel job(s).
  stop         Stop cluster(s).
  autostop     Schedule an autostop or autodown for cluster(s).
  start        Restart cluster(s).
  down         Tear down cluster(s).
  check        Check which clouds are available to use.
  show-gpus    Show supported GPU/TPU/accelerators and their prices.
  storage      SkyPilot Storage CLI.
  bench        SkyPilot Benchmark CLI.
  spot         Managed Spot CLI (spot instances with auto-recovery).
  serve        SkyServe CLI (multi-region, multi-cloud serving).
```

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [x] Any manual or new tests for this PR (please specify below): above; rendered docs
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

File: docs/source/examples/gpu-jupyter.rst [hunk 1]
```diff
@@ -5,13 +5,13 @@ Jupyter notebooks are a useful tool for interactive development, debugging, and
...
-To get a machine with a GPU attached, we recommend using an interactive **GPU node**.
-You can read more about interactive nodes :ref:`here <interactive-nodes>`.
+To get a machine with a GPU attached, use:
 
 .. code-block:: bash
 
    # Launch a VM with 1 NVIDIA GPU and forward port 8888 to localhost
-   sky gpunode -p 8888 -c jupyter-vm --gpus K80:1
+   sky launch -c jupyter-vm --gpus K80:1
+   ssh -L 8888:localhost:8888 jupyter-vm
```
Comment (Michaelvll) [inline docs/source/examples/gpu-jupyter.rst:15]:
```suggestion
Access the jupyter notebook on the browser with the url: ``http://localhost:8888``
```

File: docs/source/reference/tpu.rst [hunk 2]
```diff
@@ -48,7 +50,7 @@ More details can be found on GCP `documentation <https://cloud.google.com/tpu/do
 TPU VMs
 -------
 
-To use TPU VMs, set the following in a task YAML's ``resources`` field: 
+To use TPU VMs, set the following in a task YAML's ``resources`` field:
```
Comment (Michaelvll) [inline docs/source/reference/tpu.rst:53]: nit: in the future, we probably need to add the `--tpu-vm` for `sky launch` as well.

**Review by Michaelvll — APPROVED** [review]
Thanks for deprecating the commands @concretevitamin! LGTM.

File: docs/source/reference/tpu.rst (hunk 2 above)
Comment (concretevitamin) [reply docs/source/reference/tpu.rst:53]: Or we can switch the default from tpu nodes to tpu vms @infwinston

File: docs/source/examples/gpu-jupyter.rst (hunk 1 above)
Comment (concretevitamin) [reply docs/source/examples/gpu-jupyter.rst:15]: Actually, it's mentioned a bit down below. Reworded.

File: sky/cli.py [hunk 3]
```diff
@@ -1205,6 +1205,18 @@ def _add_command_alias_to_group(group, command, name, hidden):
...
+def _deprecate_and_hide_command(group, command_to_deprecate,
+                                alternative_command):
+    """Hide a command and show a deprecation note, hinting the alternative."""
+    command_to_deprecate.hidden = True
+    if group is not None:
+        orig = f'sky {group.name} {command_to_deprecate.name}'
+    else:
+        orig = f'sky {command_to_deprecate.name}'
+    command_to_deprecate.invoke = _with_deprecation_warning(
+        command_to_deprecate.invoke, alternative_command, orig)
```
Comment (romilbhardwaj) [inline sky/cli.py:1217]: should we consider implementing this as a method decorator?

**Review by romilbhardwaj — APPROVED** [review]
thanks @concretevitamin!

File: sky/cli.py (hunk 3 above)
Comment (concretevitamin) [reply sky/cli.py:1217]: Either way seems fine to me. This is following L1195 and https://google.github.io/styleguide/pyguide.html#217-function-and-method-decorators :)

**Comment by manglav** [comment]
https://skypilot.readthedocs.io/en/v0.2.5/examples/iterative-dev-project.html

FYI these docs still mention gpunode.

**Pull request merged**